src/components/author/author.test.ts

```diff
@@ -1,62 +1,67 @@
-        })
-      );
+test(
+  'more accessible experience for publish date',
+  withBrowser(async ({ utils, page }) => {
+    await utils.injectHTML(
+      await template({
+        // The avatar is not included because I couldn't figure out how
+        // to include it. For the purposes of this test, though, it is
+        // not important so I left it out.
+        authors: [
+          {
+            name: 'Shakira Isabel Mebarak Ripoll',
+            link: 'https://www.shakira.com/',
+          },
+        ],
+        date: new Date('March 31, 2021'),
+        date_prefix: 'Presented on',
+      })
+    );
 
-      // Confirm a more accessible user experience
-      await screen.getByText('Presented on March 31st, 2021');
-    })
-  );
+    const body = await page.evaluateHandle<ElementHandle>(() => document.body);
+    expect(await getAccessibilityTree(body)).toMatchInlineSnapshot(`
+        text "By"
+        link "Shakira Isabel Mebarak Ripoll"
+          text "Shakira Isabel Mebarak Ripoll"
+        text "Presented on March 31st, 2021"
+      `);
+  })
+);
 
-  test(
-    'meta is prioritized over date',
-    withBrowser(async ({ utils, screen }) => {
-      await utils.injectHTML(
-        await template({
-          // The avatar is not included because I couldn't figure out how
-          // to include it. For the purposes of this test, though, it is
-          // not important so I left it out.
-          authors: [
-            {
-              name: 'Shakira Isabel Mebarak Ripoll',
-              link: 'https://www.shakira.com/',
-            },
-          ],
-          date: new Date('March 31, 2021'),
-          meta: 'Singer and songwriter',
-        })
-      );
+test(
+  'meta is prioritized over date',
+  withBrowser(async ({ utils, page }) => {
+    await utils.injectHTML(
+      await template({
+        // The avatar is not included because I couldn't figure out how
+        // to include it. For the purposes of this test, though, it is
+        // not important so I left it out.
+        authors: [
+          {
+            name: 'Shakira Isabel Mebarak Ripoll',
+            link: 'https://www.shakira.com/',
+          },
+        ],
+        date: new Date('March 31, 2021'),
+        meta: 'Singer and songwriter',
+      })
+    );
 
-      // Confirm the meta value is rendered
-      await screen.getByText('Singer and songwriter');
+    const body = await page.evaluateHandle<ElementHandle>(() => document.body);
 
-      // Confirm the date is not rendered
-      const publishDate = await screen.queryByText(
-        'Published on March 31st, 2021'
-      );
-      expect(publishDate).toBeNull();
-    })
-  );
-});
+    // Confirm the meta value is rendered and the date is not rendered
+    expect(await getAccessibilityTree(body)).toMatchInlineSnapshot(`
+        text "By"
+        link "Shakira Isabel Mebarak Ripoll"
+          text "Shakira Isabel Mebarak Ripoll"
+        text "Singer and songwriter"
+      `);
```

Comment:
Nit: I wonder if it would match the original intent of this test better to do something like this?

```js
const a11yTree = await getAccessibilityTree(body));

expect(a11yTree.includes('Singer and songwriter')).toBeTrue();
expect(a11yTree.includes('Published on March 31st, 2021')).toBeFalse();
```

(This is not a blocking comment. Feel free to ignore it.)

Reply:
I think the snapshot is better because it will be more clear when it fails by showing where it specifically failed, and also it will fail in more cases (for example if things changed order or role, etc), and since it is an inline snapshot it just requires telling Jest to update it if the change is intended instead of having to update it manually.